actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -123,7 +123,7 @@ def assert_enqueued_emails(number, &block)
     #       ContactMailer.with(email: 'user@example.com').welcome.deliver_later
     #     end
     #   end
-    def assert_enqueued_email_with(mailer, method, args: nil, queue: "mailers", &block)
+    def assert_enqueued_email_with(mailer, method, args: nil, queue: ActionMailer::Base.delivery_job.new.queue_name, &block)
```

Comment:
@rafaelfranca  the problem that `ActionMailer::Base.delivery_job.queue_name` does not return the same value as `ActionMailer::Base.delivery_job.new.queue_name`. I'll handle it in separate issue

Reply:
We only care about `ActionMailer::Base.deliver_later_queue_name`. If people configured a different job with a different queue than `ActionMailer::Base.deliver_later_queue_name` they should pass the `queue:` argument explicitly.

Can you change this method to use `ActionMailer::Base.deliver_later_queue_name`?

Reply:
@rafaelfranca `ActionMailer::Base.deliver_later_queue_name` returns `mailers` instead of `default` because `action_mailer.deliver_later_queue_name = nil` does not change it to  to `default`. That what I wanted to fix in separate or. Should I change default from `action_mailer.deliver_later_queue_name = nil` to `action_mailer.deliver_later_queue_name = :default`

This will change documentation and release notes, what do you think?

Reply:
> ActionMailer::Base.deliver_later_queue_name returns mailers instead of default when action_mailer.deliver_later_queue_name = nil.

Isn't that the expected behavior. If `deliver_later_queue_name` is `nil` this method should use `:mailers`, not default. It should only use `:default` if `action_mailer.deliver_later_queue_name = :default`.

Reply:
oh. that is right, we want the opposite. But how `deliver_later_queue_name` returns `:mailers` if the value is `nil`? If should only return `:mailers` if the setter was not called.

Reply:
1. Based on https://github.com/rails/rails/issues/40730#issuecomment-737504985 I thought that you wanted to use  `default` queue instead of `mailers`
2. For the same settings `ActionMailer::Base.delivery_job.new.queue_name` returns `default` (which MailerJob use as queue name), but `ActionMailer::Base.deliver_later_queue_name` returns `mailers`

Reply:
I'll check why we have a different queue name in this issue, so then I'll fix 2 problems in one PR, if this ok

Reply:
Yes, it is fine to fix the issue in the same PR, but if we set `ActionMailer::Base.deliver_later_queue_name` to `nil` it should return `nil`.

Reply:
@rafaelfranca could you confirm that the default queue should be `default` (like it works now) (based on https://github.com/rafaelfranca/omg-rails/commit/39304496ac597c98c12981e477760f6b0e1ab04b#diff-614a0b9e89499dcd4783283bb2b1b578567aae98e5c221b5fedfd76c3a4dc1ef) instead of `mailers`.

Some thoughts:
If `ActionMailer::Base.deliver_later_queue_name => nil` then we definitely could not use it as default value for `queue: ` arg

Reply:
It should be `:defaults` only if `config.action_mailer.deliver_later_queue_name` is set to `nil` (Rails 6.1 default). If nothing is set it should be `:mailers` (Rails < 6.1 default).

> If ActionMailer::Base.deliver_later_queue_name => nil then we definitely could not use it as default value for queue: arg.

We can use queue: `queue: ActionMailer::Base.deliver_later_queue_name || "defaults"`.